packages/next/client/index.js

```diff
@@ -101,7 +101,7 @@ class Container extends React.Component {
     // If page was exported and has a querystring
     // If it's a dynamic route or has a querystring
     if (
-      data.nextExport &&
+      (data.nextExport || data.isSSG) &&
```

Comment:
This logic isn't right, SSG pages should only refresh for `location.search` -- not dynamic routes. Right?

Reply:
Updated